lib/taskana-core/src/test/java/acceptance/classification/CreateClassificationAccTest.java

```diff
@@ -1,297 +1,374 @@
-        .isInstanceOf(ClassificationAlreadyExistException.class);
-  }
+  void should_ThrowException_TryingToCreateClassificationWithInvalidType() {
+    String invalidType = "UNKNOWN_TYPE";
 
-  @WithAccessId(user = "businessadmin")
-  @Test
-  void testCreateClassificationInUnknownDomain() {
     Classification classification =
-        CLASSIFICATION_SERVICE.newClassification("Key3", "UNKNOWN_DOMAIN", "TASK");
-    assertThatThrownBy(() -> CLASSIFICATION_SERVICE.createClassification(classification))
-        .isInstanceOf(DomainNotFoundException.class);
+        classificationService.newClassification("KeyErrCreation", "DOMAIN_A", invalidType);
+
+    assertThatThrownBy(() -> classificationService.createClassification(classification))
+        .isInstanceOf(InvalidArgumentException.class)
+        .hasMessage(
+            String.format(
+                "Given classification type %s is not valid according to the configuration.",
+                invalidType));
   }
 
   @WithAccessId(user = "businessadmin")
   @Test
-  void testCreateClassificationOfUnknownType() {
+  void should_ThrowException_TryingToCreateClassificationWithInvalidCategory() {
+    String invalidCategory = "UNKNOWN_CATEGORY";
+
     Classification classification =
-        CLASSIFICATION_SERVICE.newClassification("Key3", "DOMAIN_A", "UNKNOWN_TYPE");
-    assertThatThrownBy(() -> CLASSIFICATION_SERVICE.createClassification(classification))
-        .isInstanceOf(InvalidArgumentException.class);
+        classificationService.newClassification("KeyErrCreation", "DOMAIN_A", "TASK");
+    classification.setCategory(invalidCategory);
+
+    assertThatThrownBy(() -> classificationService.createClassification(classification))
+        .isInstanceOf(InvalidArgumentException.class)
+        .hasMessage(
+            String.format(
+                "Given classification category %s with type TASK is not valid according to the"
+                    + " configuration.",
+                invalidCategory));
   }
 
   @WithAccessId(user = "businessadmin")
   @Test
-  void testCreateClassificationOfUnknownCategory() {
+  void should_ThrowException_TryingToCreateClassificationWithInvalidParentKey() {
+    String invalidParentKey = "UNKNOWN_KEY";
+
     Classification classification =
-        CLASSIFICATION_SERVICE.newClassification("Key4", "DOMAIN_A", "TASK");
-    classification.setCategory("UNKNOWN_CATEGORY");
-    assertThatThrownBy(() -> CLASSIFICATION_SERVICE.createClassification(classification))
-        .isInstanceOf(InvalidArgumentException.class);
+        classificationService.newClassification("KeyErrCreation", "", "TASK");
+    classification.setParentKey(invalidParentKey);
+
+    // Make sure that there is no (Parent) Classification with this Key.
+    List<ClassificationSummary> classifications =
+        classificationService.createClassificationQuery().keyIn(invalidParentKey).list();
+    assertThat(classifications).isEmpty();
```

Comment:
remove. This is not necessary.

Reply:
Why is this not necessary for key. If there's already a classification with this key, the purpose of the test would be missed. And I think theoretically there could be a classification with this key.